File: objective-c/difference-of-squares/dos/dos/DifferenceOfSquares.m
#import "DifferenceOfSquares.h"

@interface DifferenceOfSquares ()
@property (nonatomic, assign) NSInteger max;
@end
@implementation DifferenceOfSquares
- (instancetype)initWithMax:(NSInteger)max
{
    self = [super init];
    if (self) {
        _max = max;
    }
    return self;
}
- (NSInteger)squareOfSum
{
    NSInteger sum = 0;
    for (NSInteger i = 1; i <= _max; ++i) {
        sum += i;
    }
    return (NSInteger)pow(sum, 2);
}
- (NSInteger)sumOfSquares
{
    NSInteger sum = 0;
    for (NSInteger i = 1; i <= _max; ++i) {
        sum += pow(i, 2);
    }
    return (NSInteger)sum;
}
- (NSInteger)differenceOfSquares
{
    return _squareOfSum - _sumOfSquares;
}
@end

File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {

    let square: Int

    init(_ square: Int) {
        self.square = square
    }

    public var squareOfSum: Int {
        let s = (0...square).reduce(0, +)
        return s * s
    }

    public var sumOfSquares: Int {
        (0...square).reduce(0, { $0 + $1 * $1 })

    }

    public var differenceOfSquares: Int {
        squareOfSum - sumOfSquares
    }


}
